### PR TITLE
Core concepts

### PR DESCRIPTION
**Pull request by ryanharmanepam**

### CODE REVIEW

File: src/components/counter/index.tsx [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import React, { useState } from "react";
+import { Button } from "..";
+
+export const Counter = () => {
+  const [currentCount, setCurrentCount] = useState<number>(0);
+
+  return (
+    <div>
+      <Button onClick={() => setCurrentCount(currentCount + 1)}>Up!</Button>
+      <Button onClick={() => setCurrentCount(currentCount - 1)}>Down!</Button>
```
Comment (eduardo-neto-epam) [inline src/components/counter/index.tsx]: Avoid the use of in-line functions.

File: src/components/genreSelection/index.tsx [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  return <Tabs tabs={tabs} onTabClick={onTabClick} />;
+};
+
+const genreTabs = [
+  { id: 0, name: "All", isSelected: true },
+  { id: 1, name: "Documentary", isSelected: false },
+  { id: 2, name: "Comedy", isSelected: false },
+  { id: 3, name: "Horror", isSelected: false },
+  { id: 4, name: "Crime", isSelected: false },
+];
```
Comment (eduardo-neto-epam) [inline src/components/genreSelection/index.tsx]: Try to place mockData in its own file.

File: src/components/genreSelection/index.tsx [hunk 3]
```diff
@@ -0,0 +1,23 @@
+import React, { useState } from "react";
+import { Tabs } from "..";
+
+export const GenreSelection = () => {
+  const [tabs, setTabs] = useState(genreTabs);
+
+  const onTabClick = (id: number) => {
+    const newTabs = tabs.map((i) =>
```
Comment (eduardo-neto-epam) [inline src/components/genreSelection/index.tsx]: Avoid the use of single characters. Try to be descriptive with your variables. "i " can be easily confused with index. if you call the variable tab, readability improves.

File: src/components/genreSelection/index.tsx [hunk 4]
```diff
@@ -0,0 +1,23 @@
+import React, { useState } from "react";
```
Comment (eduardo-neto-epam) [inline src/components/genreSelection/index.tsx]: Avoid to declare variables that are not used. e.g. React
React 18 does not need the import React

File: src/components/ui/tabs/index.tsx [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+  onTabClick: (id: number) => void;
+}
+
+// TODO: Convert to nav later
+export const Tabs = ({ tabs, onTabClick }: TabsProps) => {
+  return (
+    <div>
+      {/* TODO: Styling for selected tab */}
+      {tabs.map((i) => (
+        <div onClick={() => onTabClick(i.id)}>{i.name}</div>
```
Comment (eduardo-neto-epam) [inline src/components/ui/tabs/index.tsx]: I suggest the creation of a Tab component as a UI element. Tabs is more of a container/layout.
Make sure when using map to add a key prop to the children.
Also, consider something like this to see what tab is active:
`      {tabs.map(({id, name, isSelected}) => (
        <div key={id} onClick={() => onTabClick(id)}>{name} - {isSelected && 'ACTIVE'}</div>
      ))}`

File: src/components/ui/button/index.tsx [hunk 6]
```diff
@@ -0,0 +1,10 @@
+import React from "react";
+
+interface ButtonProps {
```
Comment (eduardo-neto-epam) [inline src/components/ui/button/index.tsx]: A button component should have the same props as a native html button.
ref: https://dev.indooroutdoor.io/how-to-extend-any-html-element-with-react-and-typescript
https://react-typescript-cheatsheet.netlify.app/docs/advanced/patterns_by_usecase/#wrappingmirroring-a-html-element

File: src/components/ui/input/index.tsx [hunk 7]
```diff
@@ -0,0 +1,5 @@
+import React from "react";
+
```
Comment (eduardo-neto-epam) [inline src/components/ui/input/index.tsx:5]: See https://dev.to/giselamd/creating-a-react-input-component-in-typescript-hai for a generic Input well typed.

File: src/components/ui/input/index.tsx [hunk 8]
```diff
@@ -0,0 +1,5 @@
+import React from "react";
```
Comment (eduardo-neto-epam) [inline src/components/ui/input/index.tsx]: No need for this in React 18

**Review by eduardo-neto-epam — CHANGES_REQUESTED** [review]
Remember that if you're using React 18, to use its features.
Make sure to do your best to provide a clean and optimised code.
Congratulations on using Typescript, that said, make sure elements are properly typed.

File: src/components/ui/tabs/index.tsx (hunk 5 above)
Comment (ryanharmanepam) [reply src/components/ui/tabs/index.tsx]: Tab component has been created

File: src/components/searchForm/index.tsx [hunk 9]
```diff
@@ -0,0 +1,14 @@
...
+
+export const SearchForm = () => {
+  const onSearch = () => {
+    // TODO: Finalise search action w/ api call / filter
+  };
+
+  return (
+    <div>
+      <Input />
+      <Button onClick={() => onSearch()}>Search</Button>
```
Comment (eduardo-neto-epam) [inline src/components/searchForm/index.tsx:11]: No need to pass a callback to call onSearch if you have no arguments to pass to the function.

**Review by eduardo-neto-epam — APPROVED** [review]
Well done,  although onSearch got away from being fixed.

**Pull request merged**